### ISSUE TITLE
Installing on OSX with brew-installed postgres 15

### ISSUE THREAD

**Issue by symbolicsorcerer** [pgvector/pgvector#262]
What am I missing here? 🤔 

```
➜  pgvector git:(v0.5.0) export PG_CONFIG=$(which pg_config)                                                                    
➜  pgvector git:(v0.5.0) echo $PG_CONFIG                                                                                        
/opt/homebrew/bin/pg_config                                                                                                     
➜  pgvector git:(v0.5.0) make install                                                                                           
/bin/sh /opt/homebrew/lib/postgresql@14/pgxs/src/makefiles/../../config/install-sh -c -d '/opt/homebrew/lib/postgresql@14'      
/bin/sh /opt/homebrew/lib/postgresql@14/pgxs/src/makefiles/../../config/install-sh -c -d '/opt/homebrew/share/postgresql@14/exte
nsion'                                                                                                                          
/bin/sh /opt/homebrew/lib/postgresql@14/pgxs/src/makefiles/../../config/install-sh -c -d '/opt/homebrew/share/postgresql@14/exte
nsion'                                                                                                                          
/usr/bin/install -c -m 755  vector.so '/opt/homebrew/lib/postgresql@14/vector.so'                                               
/usr/bin/install -c -m 644 .//vector.control '/opt/homebrew/share/postgresql@14/extension/'                                     
/usr/bin/install -c -m 644 .//sql/vector--0.1.0--0.1.1.sql .//sql/vector--0.1.1--0.1.3.sql .//sql/vector--0.1.3--0.1.4.sql .//sq
l/vector--0.1.4--0.1.5.sql .//sql/vector--0.1.5--0.1.6.sql .//sql/vector--0.1.6--0.1.7.sql .//sql/vector--0.1.7--0.1.8.sql .//sq
l/vector--0.1.8--0.2.0.sql .//sql/vector--0.2.0--0.2.1.sql .//sql/vector--0.2.1--0.2.2.sql .//sql/vector--0.2.2--0.2.3.sql .//sq
l/vector--0.2.3--0.2.4.sql .//sql/vector--0.2.4--0.2.5.sql .//sql/vector--0.2.5--0.2.6.sql .//sql/vector--0.2.6--0.2.7.sql .//sq
l/vector--0.2.7--0.3.0.sql .//sql/vector--0.3.0--0.3.1.sql .//sql/vector--0.3.1--0.3.2.sql .//sql/vector--0.3.2--0.4.0.sql .//sq
l/vector--0.4.0--0.4.1.sql .//sql/vector--0.4.1--0.4.2.sql .//sql/vector--0.4.2--0.4.3.sql .//sql/vector--0.4.3--0.4.4.sql .//sq
l/vector--0.4.4--0.5.0.sql .//sql/vector--0.5.0.sql  '/opt/homebrew/share/postgresql@14/extension/'                             
/bin/sh /opt/homebrew/lib/postgresql@14/pgxs/src/makefiles/../../config/install-sh -c -d '/opt/homebrew/include/postgresql@14/se
rver/extension/vector/'                                                                                                         
/usr/bin/install -c -m 644   .//src/vector.h '/opt/homebrew/include/postgresql@14/server/extension/vector/' 
```

**Comment by symbolicsorcerer** [comment]
Managed to have postgresql 14 installed as well, and pgvector via brew because I was trying to get it working. Some shell history:

```
$ brew remove pgvector
$ brew remove postgresql@14
$ echo 'export PATH="/opt/homebrew/opt/postgresql@15/bin:$PATH"' >> ~/.zshrc
$ source ~/.zshrc
$ which pg_config                                  
/opt/homebrew/opt/postgresql@15/bin/pg_config
$ export PG_CONFIG=$(which pg_config)
$ make install
```

Now I'm at:

```
postgres=# create extension vector;
ERROR:  incompatible library "/opt/homebrew/opt/postgresql@15/lib/postgresql/vector.so": version mismatch
```

**Comment by jkatz** [comment]
@symbolicsorcerer The above is a bit hard to follow, but once you have your build/install environment sorted out, I'd recommend reinstalling:

```
make uninstall # or sudo make uninstall based upon the path
make clean
make
make install # or sudo make install based upon the path
```

**Comment by ankane** [comment]
A `version mismatch` error means the extension was compiled against a different version of Postgres (the error `DETAIL:` will show which version). Running `make clean` as @jkatz mentioned will fix it.

https://github.com/pgvector/pgvector#postgres-location
